File: VisualStudio/VBasic/w7cpframe/clsPfConnectivity.vb
'
' Copyright (c)2016 S C Harrison
' Refer to License.txt for terms and conditions of use.
'
Option Explicit On
Public Class clsPfConnectivity

#Const isUsingExcel = 0

    Public key As Integer

    Public jj As Byte            '.. joint No. @ end "j" of a member ..  [na]
    Public jk As Byte            '.. joint No. @ end "k" of a member ..  [nb]
    Public sect As Byte          '.. section group of member ..          [ns]
    Public rel_i As Byte         '.. end i release of member ..          [mra]
    Public rel_j As Byte         '.. end j release of member ..          [mrb]
    Public jnt_jj As clsPfForce
    Public jnt_jk As clsPfForce

    Sub initialise()
        jj = 0
        jk = 0
        sect = 0
        rel_i = 0
        rel_j = 0

        jnt_jj = New clsPfForce
        jnt_jj.initialise()

        jnt_jk = New clsPfForce
        jnt_jk.initialise()

    End Sub

    Sub setValues(ByVal memberKey As Integer, ByVal NodeA As Integer, ByVal NodeB As Integer, ByVal sectionKey As Integer, ByVal ReleaseA As Integer, ByVal ReleaseB As Integer)
        key = memberKey
        jj = NodeA
        jk = NodeB
        sect = sectionKey
        rel_i = ReleaseA
        rel_j = ReleaseB
    End Sub

    Function sprint() As String
        sprint = StrLPad(Format(key, "##0"), 8) _
                 & StrLPad(Format(jj, "##0"), 6) _
                 & StrLPad(Format(jk, "##0"), 6) _
                 & StrLPad(Format(sect, "##0"), 6) _
                 & StrLPad(Format(rel_i, "##0"), 6) _
                 & StrLPad(Format(rel_j, "##0"), 2)

    End Function


    Sub cprint()

        Console.WriteLine(sprint)
        '  jnt_jj.cprint
        '  jnt_jk.cprint

    End Sub

    Sub fprint(ByVal fp As Integer)
        Print(fp, sprint)
    End Sub


#If isUsingExcel = 1 Then

    Sub wbkPrint(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            .Offset(RecNum, 0).Value = key
            .Offset(RecNum, 1).Value = jj
            .Offset(RecNum, 2).Value = jk
            .Offset(RecNum, 3).Value = sect
            .Offset(RecNum, 4).Value = rel_i
            .Offset(RecNum, 5).Value = rel_j
        End With
    End Sub

    Sub wbkRead(ByVal dataRec As Range, ByVal RecNum As Byte)
        With dataRec
            key = .Offset(RecNum, 0).Value
            jj = .Offset(RecNum, 1).Value
            jk = .Offset(RecNum, 2).Value
            sect = .Offset(RecNum, 3).Value
            rel_i = .Offset(RecNum, 4).Value
            rel_j = .Offset(RecNum, 5).Value
        End With
    End Sub

#End If

End Class

File: Eclipse/w7cpframe/src/clsPfConnectivity.java
//
// Copyright (c)2016 S C Harrison
// Refer to License.txt for terms and conditions of use.
//
import java.io.*;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

    class clsPfConnectivity
    {
        //Define Class

        public int key = 0;

        public int jj = 0;            //.. joint No. @ end "j" of a member ..  [na]
        public int jk = 0;            //.. joint No. @ end "k" of a member ..  [nb]
        public int sect = 0;          //.. section group of member ..          [ns]
        public int rel_i = 0;         //.. end i release of member ..          [mra]
        public int rel_j = 0;         //.. end j release of member ..          [mrb]

        public double L = 0;             //Length of Member

        public clsPfForce jnt_jj; //= null; //new clsPfForce; // clsPfForce
        public clsPfForce jnt_jk; //= null; //new clsPfForce; // clsPfForce


        

        public void initialise()
        {
            key = 0;
            jj = 0;
            jk = 0;
            sect = 0;
            rel_i = 0;
            rel_j = 0;

            //  this.jnt_jj = new clsPfForce;
            //  this.jnt_jj.initialise;
            //  
            //  this.jnt_jk = new clsPfForce;
            //  this.jnt_jk.initialise;

        }

        public void setValues(int memberKey, int NodeA, int NodeB, int sectionKey, int ReleaseA, int ReleaseB)
        {
            key = memberKey;
            jj = NodeA;
            jk = NodeB;
            sect = sectionKey;
            rel_i = ReleaseA;
            rel_j = ReleaseB;
        }

        public String sprint()
        {
            String s;

            s = "";
            s = s + String.format("%8d",key);
            s = s + String.format("%8d",jj);
            s = s + String.format("%8d",jk);
            s = s + String.format("%8d",sect);
            s = s + String.format("%8d",rel_i);
            s = s + String.format("%8d",rel_j);

            return s;

        }

        public void cprint()
        {
            System.out.println(sprint());
            //  jnt_jj.cprint
            //  jnt_jk.cprint

        }

        public void fprint(BufferedWriter fp)
        {
            try {
				fp.write(sprint());
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
        }

        public void fgetData(BufferedReader fp)
        {
            String s="";

            System.out.println("fgetData ...");

            try {
				s = fp.readLine();
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
            sgetData(s);

            System.out.println("... fgetData");
        }

        public void sgetData(String s)
        {
            String[] dataflds = new String[10]; //(0 To 9);
            int i;

            System.out.println("sgetData ...");
            System.out.println(s);

            //Trim Trailing Spaces from input String
            String regExpTrimPattern = "^\\s+|\\s+$"; //trim trailing spaces          
            Pattern regExpPatternObj = Pattern.compile(regExpTrimPattern);
            Matcher regExpMatches = regExpPatternObj.matcher(s);
            String regResult = regExpMatches.replaceAll("");
            
            //Get Data fields from input String
            String regExpNumberPattern = "-?\\d+(?:[,.]\\d+)?";
            regExpPatternObj = Pattern.compile(regExpNumberPattern);
            regExpMatches = regExpPatternObj.matcher(regResult);
            i=0;
            while (regExpMatches.find()){
            	dataflds[i] = regExpMatches.group().trim();
            	i = i+1;
            }


   
            key = Integer.parseInt(dataflds[0]);
            jj = Integer.parseInt(dataflds[1]);
            jk = Integer.parseInt(dataflds[2]);
            sect = Integer.parseInt(dataflds[3]);
            rel_i = Integer.parseInt(dataflds[4]);
            rel_j = Integer.parseInt(dataflds[5]);

            System.out.println("... sgetData");
        }



    } //class